src/vehicle.cpp

```diff
@@ -1316,41 +1310,108 @@ void vehicle::use_controls()
     }
 }
 
-void vehicle::start_engine()
+bool vehicle::start_engine( const int e )
+{
+    if( !is_engine_on( e ) ) { return false; }
+
+    const vpart_info &einfo = part_info( engines[e] );
+
+    if( !fuel_left( einfo.fuel_type ) ) {
+        add_msg( _("The %s is out of %s."), einfo.name.c_str(), ammo_name( einfo.fuel_type ).c_str() );
+        return false;
+    }
+
+    const bool combustion = einfo.fuel_type == fuel_type_gasoline ||
+        einfo.fuel_type == fuel_type_diesel;
+    const double dmg = 1 - (parts[engines[e]].hp / einfo.durability);
```

Comment:
Why a `double` if you're doing integer division? You have to cast one of the involved variables if you want a floating point result. This issue occurs in a number of other places as well.